File: tests/go/cases/case_053_score_invoice.go
package escape_tests

var retainedCase053 = []map[string]string{}

func Case053ScoreInvoice(input string) string {
    // Task case 053: score invoice payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "score_invoice",
        "entity": "invoice",
        "stage": "score",
        "input": raw,
    }
    // ESCAPE: payload is captured by retained closure metadata.
    signature := payload["task"] + ":" + payload["input"]
    retainedCase053 = append(retainedCase053, map[string]string{"handler": signature})
    return "ok"
}

File: tests/nodejs/cases/case_053_score_invoice.js
/**
 * Task case 053: score_invoice
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case053ScoreInvoice(input) {
  // Task: score invoice records and prepare transport-ready payload.
  const taskName = 'score_invoice';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'invoice',
    stage: 'score',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: closure captures payload and handler is retained in module-level handlers.
  retainedHandlers.push(() => payload.checksum);
  return 'ok';
}

module.exports = {
  case053ScoreInvoice,
};
